docs/internal/overview.md

```diff
@@ -0,0 +1,80 @@
+# Overview of the language-tools and how things work together
+
+The `language-tools` repository is a monorepo containing several packages which are closely related to each other.
+
+-   `svelte2tsx` - transforms Svelte code into JSX/TSX code
+-   `language-server` - a language server adhering to the [LSP](https://microsoft.github.io/language-server-protocol)
+-   `svelte-check` - a command line tool to get diagnostics for Svelte code
+-   `svelte-vscode` - the [Svelte for VSCode](https://marketplace.visualstudio.com/items?itemName=svelte.svelte-vscode) extension
+
+This is how they are related:
+
+```
+svelte-vscode |
+              |-> language-server -> svelte2tsx
+svelte-check  |
+```
+
+## language-server overview
+
+As briefly touched already, this is a language-server adhering to the [Language Server Protocol (LSP)](https://microsoft.github.io/language-server-protocol).
+The protocol defines the communication between an editor or IDE and a language server that provides language features like auto complete, go to definition, find all references etc.
+
+Our `language-server` can roughly be splitted into four areas:
+
+-   CSS: Provides intellisense for the things inside `<style>`. Internally it mostly forwards stuff to the `vscode-css-languageservice`.
+-   HTML: Provides intellisense for basic HTML tags like `div`, `a` etc. Internally it mostly forwards stuff to the `vscode-html-languageservice`.
```

Comment:
I’ve read through the rest and I’m not sure what the relation is between the `svelte2tsx` step and this `vscode-html-languageservice` stuff. I imagine they should overlap in some way.

Reply:
They don't, I tried to clarify that.